sdk/logical/audit.go

```diff
@@ -20,3 +20,36 @@ type MarshalOptions struct {
 type OptMarshaler interface {
 	MarshalJSONWithOptions(*MarshalOptions) ([]byte, error)
 }
+
+// LogInputBexpr is used for evaluating boolean expressions with go-bexpr.
+type LogInputBexpr struct {
+	MountPoint string `bexpr:"mount_point"`
+	MountType  string `bexpr:"mount_type"`
+	Namespace  string `bexpr:"namespace"`
+	Operation  string `bexpr:"operation"`
+	Path       string `bexpr:"path"`
+}
+
+// BexprDatum returns values from a LogInput formatted for use in evaluating go-bexpr boolean expressions.
+// The namespace should be supplied from the current request's context.
+func (l *LogInput) BexprDatum(namespace string) any {
```

Comment:
why does this return `any`? Can it be typed?

Reply:
Because The Bexpr evaluator wants `any` and we followed the pattern that has been created by the eventbus:
https://github.com/hashicorp/vault/blob/main/sdk/logical/events.go#L80-L105

Reply:
Sorry, to answer the question though, yep, you're right. It's been updated.